playbooks/robusta_playbooks/pvc_snapshots.py

```diff
@@ -0,0 +1,55 @@
+from pydantic import SecretStr
+from kubernetes import client
+from robusta.api import *
+
+
+class VolumeSnapshotParams(ActionParams):
+    """
+    :var name: name of the PersistentVolumeClaim to backup
+    :var namespace: namespace of the PersistentVolumeClaim to backup
+    :example name: some_pvc_name
+    """
+
+    name: str
+    namespace: str = "default"
+
+
+@action
+def create_pvc_snapshot(event: ScheduledExecutionEvent, params: VolumeSnapshotParams):
+    """
+    Backup a PersistentVolume by creating a VolumeSnapshot
+    """
+
+    pvc = PersistentVolumeClaim().read(name=params.name, namespace=params.namespace)
```

Comment:
Do we really need to read it?
We use only the name & namespace we have in the params.
Is this an implicit check that the PVC exists (if it doesn't an exception is thrown, right?)
Should we handle the case it doesn't exist cleaner?

Reply:
Yes, it's only to check that it exists. How do you recommend handling it?

Reply:
Maybe we should catch the exception and fail nicely.
Should we notify the user the backup has failed using a finding?

Reply:
If the any action fails will add some generic finding saying there was an error.
Here, you can send something more user friendly, saying "Backup failed because volume xyz doesn't exist"
Maybe even add the same aggregation key.
Not a must obviously, but it might make it easier to look for such errors in the UI

Reply:
Pushed a fix.